src/commands/schemas.ts

```diff
@@ -217,6 +222,107 @@ async function evolveSchemaSubjectCommand(subject: Subject) {
   await evolveSchemaCommand(schema);
 }
 
+/**
+ * Delete a single schema version.
+ *
+ * If was the last version bound to the subject, the subject will disappear also.
+ *
+ */
+async function deleteSchemaVersionCommand(schema: Schema) {
+  if (!(schema instanceof Schema)) {
+    logger.error("deleteSchemaVersionCommand called with invalid argument type", schema);
+    return;
+  }
+
+  const loader = ResourceLoader.getInstance(schema.connectionId);
+  const schemaGroup = await loader.getSchemasForSubject(schema.environmentId!, schema.subject);
+
+  // Ensure is still present in the registry / UI view gestured from wasn't stale.
+  const found = schemaGroup.find((s) => s.id === schema.id) !== undefined;
+  if (!found) {
+    vscode.window.showErrorMessage(`Schema not found in registry. View needs refresh?`);
+    logger.error(
+      `Schema version ${schema.version} not found in registry, cannot delete.`,
+      schemaGroup,
+    );
+    return;
+  }
+
+  // Determine if user wants to hard or soft delete.
+  const hardDelete = await hardDeletionQuickPick();
+  if (hardDelete === undefined) {
+    vscode.window.showErrorMessage("Schema deletion canceled.");
+    logger.info("User canceled schema version deletion.");
```

Comment:
I'd skip the error notification for something like this since it isn't anything the user did wrong / any problem with the delete call
```suggestion
    logger.debug("User canceled schema version deletion.");
```